test/jdk/java/nio/channels/DatagramChannel/Loopback.java

```diff
@@ -160,7 +161,7 @@ static void test(ProtocolFamily family, InetAddress group, NetworkInterface ni)
                         if (sender != null) {
                             System.out.format("received %s from %s%n", dst, sender);
                             senderPort = ((InetSocketAddress) sender).getPort();
-                            assertTrue(senderPort != localPort, "Unexpected message");
+                            assertTrue(senderPort != localPort, "Unexpected message: localPort=" + localPort);
```

Comment:
I wonder if we should add some entropy to the message content and discard any message that we receive whose payload doesn't match what we expect.

Reply:
Do you mean like this:
-            ByteBuffer src = ByteBuffer.wrap("hello".getBytes("UTF-8"));
+            String str = "hello integer" + new Random().nextInt(10000);
+            ByteBuffer src = ByteBuffer.wrap(str.getBytes("UTF-8"));

And I don't known how to discard meaasge doesn't match what we expect.

Reply:
Yes - something like that. System.nanoTime() could work too.
By discard, I mean compare the content of `dst` with what was put in `src` - and if it doesn't match, clear `dst` and continue to go back to select() without even looking at the sender port. You may want to adjust the timeout you pass to select the second time around.
Rationale is: if the payloads do not match send we're just receiving some stray packet probably sent by another (test?) process.

Reply:
Seems that it doesn't need to adjust the timeout to select the second time around.